src/utils/index.ts

```diff
@@ -255,3 +254,14 @@ export const localStorageHelper = () => {
     },
   };
 };
+
+export function downloadFileWithUrl(url?: string | null) {
+  let safeURL = url;
+  if (safeURL) {
+    if (!safeURL.startsWith('http://') && !safeURL.startsWith('https://')) {
+      safeURL = 'https://' + safeURL;
+    }
+    window.open(safeURL, '_blank', 'noopener,noreferrer');
+  }
+  return Promise.resolve();
```

Comment:
Any reason we return this?

Reply:
its just dummy code because the prop expects return type to be Promise<void>

Reply:
tip.

```ts
function downloadFileWithUrl() {
  // ...
  return Promise.resolve();
}

// ⬇️⬇️⬇️

async function downloadFileWithUrl() {
  // ...
}
```

Reply:
@bang9 ah that is better thanks ill appy this